Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -715,19 +648,37 @@ namespace AzToolsFramework
                 const auto valueAtSubPath = GetDPE()->GetAdapter()->GetContents()[subPath];
 
                 // check if it's a PropertyHandler; if it is, just set it from the DOM directly
-                auto foundEntry = m_widgetToPropertyHandler.find(childWidget);
-                if (foundEntry != m_widgetToPropertyHandler.end())
+                auto foundEntry = m_widgetToPropertyHandlerInfo.find(childWidget);
+                if (foundEntry != m_widgetToPropertyHandlerInfo.end())
                 {
-                    foundEntry->second->SetValueFromDom(valueAtSubPath);
+                    auto handlerId = AZ::Interface<PropertyEditorToolsSystemInterface>::Get()->GetPropertyHandlerForNode(valueAtSubPath);
+
+                    // check if this patch has morphed the PropertyHandler into a different type
+                    if (handlerId != foundEntry->second.handlerId)
+                    {
+                        // CreateWidgetForHandler will add a new entry to m_widgetToPropertyHandlerInfo, kill the old entry
+                        GetDPE()->ReleaseHandler(AZStd::move(foundEntry->second.hanlderInterface));
+                        m_widgetToPropertyHandlerInfo.erase(foundEntry);
+
+                        // Replace the existing handler widget with one appropriate for the new type
+                        auto replacementWidget = CreateWidgetForHandler(handlerId, valueAtSubPath);
+                        AddColumnWidget(replacementWidget, childIndex, valueAtSubPath);
+                        AddDomChildWidget(childIndex, replacementWidget);
+                    }
+                    else
+                    {
+                        // handler is the same, set the existing handler with the new value
+                        foundEntry->second.hanlderInterface->SetValueFromDom(valueAtSubPath);
+                    }
                 }
                 else
                 {
-                    QLabel* changedLabel = qobject_cast<QLabel*>(childWidget);
+                    auto changedLabel = qobject_cast<AzQtComponents::ElidingLabel*>(childWidget);
                     AZ_Assert(changedLabel, "not a label, unknown widget discovered!");
                     if (changedLabel)
                     {
                         auto labelString = AZ::Dpe::Nodes::Label::Value.ExtractFromDomNode(valueAtSubPath).value_or("");
-                        changedLabel->setText(QString::fromUtf8(labelString.data(), aznumeric_cast<int>(labelString.size())));
+                        changedLabel->SetText(QString::fromUtf8(labelString.data(), aznumeric_cast<int>(labelString.size())));
```

Comment:
😞 that's a bit of an API footgun...

Reply:
meh, the lowercase setText does exist on elided labels, but all it does is mask (not override) the QLabel version. I'm just calling the uppercase version here for clarity. The real gotcha was realizing that the QLabel version of setText, on an elided label, does not set the text.